src/state/application/updater.ts

```diff
@@ -33,11 +69,13 @@ export default function Updater(): null {
 
   useQueryCacheInvalidator()
 
+  const [msSinceLastBlock, setMsSinceLastBlock] = useState(0)
   const blockNumberCallback = useCallback(
     (blockNumber: number) => {
       setState((state) => {
         if (chainId === state.chainId) {
           if (typeof state.blockNumber !== 'number') return { chainId, blockNumber }
+          setMsSinceLastBlock(0)
```

Comment:
i wonder if we're overloading the meaning of `blockNumberCallback` here. Can you not subscribe to the redux store (`useBlockNumber`) and update `msSinceLastBlock` in there? I think that's a more 'redux-y' way of doing things.